### PR TITLE
fix: allow gosc to set default gateway on more than one network adapter

### LINKED ISSUE #1232: `ipv4_gateway` address attached to only a single `network_interface` when multiple attached to the same `r/network`
### Terraform Version
Terraform v0.13.2

### vSphere Provider Version
1.24.0

### Affected Resource(s)

- `vsphere_virtual_machine`

### Terraform Configuration Files

```hcl

locals {
  resource_pool_name = "${var.host_ip}/Resources"
}

data "vsphere_datacenter" "dc" {
  name = var.datacenter
}

data "vsphere_host" "host" {
  name          = var.host_ip
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_datastore" "datastore" {
  name          = var.datastore
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_resource_pool" "pool" {
  name          = local.resource_pool_name
  datacenter_id = data.vsphere_datacenter.dc.id
}

# PortGroup one - have ensured these work
data "vsphere_network" "portgroup_one" {
  name          = var.portgroup_one_name
  datacenter_id = data.vsphere_datacenter.dc.id
}

# PortGroup two - have ensured these work
data "vsphere_network" "portgroup_two" {
  name          = var.portgroup_two_name
  datacenter_id = data.vsphere_datacenter.dc.id
}

data "vsphere_virtual_machine" "template" {
  name          = var.template_name
  datacenter_id = data.vsphere_datacenter.dc.id
}

resource "vsphere_virtual_machine" "vm" {
  name             = var.name
  resource_pool_id = data.vsphere_resource_pool.pool.id
  datastore_id     = data.vsphere_datastore.datastore.id
  host_system_id   = data.vsphere_host.host.id
  firmware         = "efi"

  num_cpus               = 8
  cpu_hot_add_enabled    = true
  memory                 = 32768
  memory_hot_add_enabled = true

  guest_id  = data.vsphere_virtual_machine.template.guest_id
  scsi_type = data.vsphere_virtual_machine.template.scsi_type

  # NIC one
  network_interface {
    network_id   = data.vsphere_network.portgroup_one.id
    adapter_type = "vmxnet3"
  }

  # NIC two
  network_interface {
    network_id   = data.vsphere_network.portgroup_two.id
    adapter_type = "vmxnet3"
  }

  disk {
    label            = "disk-${var.name}"
    size             = data.vsphere_virtual_machine.template.disks.0.size
    thin_provisioned = true
  }

  clone {
    template_uuid = data.vsphere_virtual_machine.template.id

    customize {

      windows_options {
        computer_name  = var.name
        admin_password = var.admin_password
      }

      # NIC 1
      network_interface {
        ipv4_address    = var.nic_one_ip_address
        ipv4_netmask    = var.nic_one_ip_mask
        dns_server_list = var.dns_servers
        dns_domain      = var.domain_name
      }

      # NIC 2
      network_interface {
        ipv4_address    = var.nic_two_ip_address
        ipv4_netmask    = var.nic_two_ip_mask
        dns_server_list = var.dns_servers
        dns_domain      = var.domain_name
      }

      ipv4_gateway = var.local_gateway_ip
    }
  }
}


```

### Debug Output

https://gist.github.com/wsf11/7950094d1ec67e91c61d62bab0ca2caf

Completes normally, no errors

### Expected Behavior

I expect that two NICs are created, with two separate IP addresses, in the same subnet, both having the default gateway attached to the NIC. 

### Actual Behavior

Both NICs are created, but only one NIC has the default gateway attached. The gateway appears to be randomly assigned to one NIC or the other. Therefore, I cannot reach the VM through the NIC that does not have the gateway attached. 

I remote into the machine and run `ipconfig` to see that only one NIC, the `Ethernet0`, has the default gateway IP address correct. It is empty for the other NIC, `Ethernet 1`

### Steps to Reproduce

terraform init & apply

### Important Factoids

I have reproduced this in the vSphere client using `VM Customization Specifications` and it worked fine. The template used was created using two NICs. I do customize the NICs and change the types as part of the deployment. This is a Windows 2019 Image. Also failing for Windows 2016

The terraform state file reports the `default_ip_address` to be the IP address that doesn't have the gateway attached. I read that the `default_ip_address` is the first IP address that the gateway can communicate with? Is something being overwritten?

I put the DNS Server IP addresses into the NIC, but do not domain join yet as I domain join later. The IP addresses being used live inside a subnet and are available. 

### References

* #0000

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### PR DESCRIPTION
**Pull request by spacegospod**
### Description

It appears that `customizations_helper.go` intentionally skips the gateway settings for every adapter after the first one.
This leads to a bug where cloning a VM with multiple adapters ends up with potentially incorrect network settings.

My fix is to simply delete the code which explicitly omits the settings for subsequent adapters during GOS customization.

### Acceptance tests

- [X] Have you added an acceptance test for the functionality being added?
- [X] Have you run the acceptance tests on this branch?

I have added a new testcase that can be used to validate this fix.
The test is intentionally marked with `t.Skip()` so that it does get included in the CI runs.
This is so because it requires a Sysprepped Windows VM template.

I tested this fix by running the new test, logging into the VM and confirming that the gateway is applied to both adapters.

### Release Note

`resource/virtual_machine`: Allow `ipv4_gateway` to be set on more than one network adapter when cloning a virtual machine.

### References

Closes #1232

### CODE REVIEW

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 1]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  name             = "testacc-test"
+  resource_pool_id = "${vsphere_resource_pool.pool1.id}"
+  datastore_id     = vsphere_nas_datastore.ds1.id
+
+  num_cpus = 4
+  memory   = 4096
+  guest_id = "${data.vsphere_virtual_machine.template.guest_id}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5723]:
```suggestion
  guest_id = data.vsphere_virtual_machine.template.guest_id
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 2]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+  firmware = "efi"
+
+  wait_for_guest_net_timeout = 0
+  wait_for_guest_ip_timeout = 0
+  wait_for_guest_net_routable = false
+
+  network_interface {
+    network_id = "${data.vsphere_network.network1.id}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5731]:
```suggestion
    network_id = data.vsphere_network.network1.id
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 3]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+    adapter_type = "vmxnet3"
+  }
+
+  network_interface {
+    network_id = "${data.vsphere_network.network1.id}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5736]:
```suggestion
    network_id = data.vsphere_network.network1.id
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 4]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+    adapter_type = "vmxnet3"
+  }
+
+  disk {
+    label            = "disk0"
+    size             = "${data.vsphere_virtual_machine.template.disks.0.size}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5742]:
```suggestion
    size             = data.vsphere_virtual_machine.template.disks.0.size
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 5]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+    eagerly_scrub    = "${data.vsphere_virtual_machine.template.disks.0.eagerly_scrub}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5743]:
```suggestion
    eagerly_scrub    = data.vsphere_virtual_machine.template.disks.0.eagerly_scrub
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 6]
```diff
@@ -5677,6 +5702,81 @@ resource "vsphere_virtual_machine" "vm" {
...
+    thin_provisioned = true
+  }
+
+  clone {
+    template_uuid = "${data.vsphere_virtual_machine.template.id}"
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:5748]:
```suggestion
    template_uuid = data.vsphere_virtual_machine.template.id
```

File: vsphere/resource_vsphere_virtual_machine_test.go [hunk 7]
```diff
@@ -1643,6 +1643,31 @@ func TestAccResourceVSphereVirtualMachine_cloneModifyDiskAndSCSITypeAtSameTime(t
 	})
 }
 
+func TestAccResourceVSphereVirtualMachine_cloneMultiNICWithSameGateway(t *testing.T) {
+	// This test is intentionally disabled
+	// To run it you need to configure a Windows VM to use as a template and then
+	// run Sysprep on it.
```
Comment (tenthirtyam) [inline vsphere/resource_vsphere_virtual_machine_test.go:1649]:
```suggestion
	// This test is intentionally disabled by default.
	// Requires a Windows virtual machine template to run this test.
```

**Comment by spacegospod** [comment]
> @spacegospod - I added some minor nits to remove the legacy interpolation, but other wise this looks good and the loginc makes sense for NICs on the same network.

no problem, the rest of the VM test file is full of interpolation strings but I'm not brave enough to touch all of the tests in this change. I just fixed my own

**Review by tenthirtyam — APPROVED** [review]
lgtm!

**Pull request merged**